### PR TITLE
Glacambre's suggestion for races

### PR DESCRIPTION
**Pull request by bovine3dom**
@glacambre in my testing this doesn't work. It could be an implementation issue but I don't see any.

```
tnews.bbc.co.uk<Enter>
d
tne<Tab><Enter> # type this as quickly as you can
```

On #3196 this opens news.bbc.co.uk, on this PR it searches Google for "ne".

### CODE REVIEW

File: src/commandline_frame.ts [hunk 1]
```diff
@@ -200,14 +200,9 @@ commandline_state.clInput.addEventListener(
                 const [funcname, ...args] = response.value.slice(3).split(/\s+/)
 
                 QUEUE[QUEUE.length - 1].then(() =>
```
Comment (glacambre) [inline src/commandline_frame.ts:202]:
```suggestion
                QUEUE.push(QUEUE[QUEUE.length - 1].finally(() =>
```

File: src/commandline_frame.ts [hunk 2]
```diff
@@ -262,10 +255,7 @@ commandline_state.clInput.addEventListener("input", () => {
         if (exstr !== commandline_state.clInput.value) return
 
         onInputPromise = refresh_completions(exstr)
```
Comment (glacambre) [inline src/commandline_frame.ts:257]:
```suggestion
        onInputPromise = QUEUE[QUEUE.length - 1].finally(() => refresh_completions(exstr))
```

**Review by glacambre — COMMENTED** [review]
My idea was more like this but I'm not surprised if it won't work - I think Tridactyl's code is beyond my ability to reason about it now.

**Comment by bovine3dom** [comment]
I've been trying to reason through your version but I give up. I'll just see if it works...

**Pull request merged**